### PR TITLE
Add OpenStack Keystone Domain module

### PR DESCRIPTION
**Pull request by emonty**

### CODE REVIEW

**Comment by bcoca** [comment]
@emonty anything you need to do on this before we review? asking since you added needs_revision tag.

**Comment by gregdek** [comment]
He's still working on it.

File: cloud/openstack/os_keystone_domain.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a domain
+- os_keystone_domain: name=demo description="Demo Domain"
+'''
+
+
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_domain.py:122]: Please document return data.

File: cloud/openstack/os_keystone_domain.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        description=dict(default=None),
+        enabled=dict(default=True, type='bool'),
+        state=dict(default='present', choices=['absent', 'present']),
+        # Override endpoint_type default since this is an admin function
+        endpoint_type=dict(
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_domain.py]: Is this meant to be undocumented?

File: cloud/openstack/os_keystone_domain.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        domain = cloud.get_identity_domain(name=name)
+
+        if state == 'present':
+            if domain is None:
+                domain = cloud.create_domain(
+                    name=name, description=description, enabled=enabled)
+                changed = True
+            else:
+                if (domain.name != name or domain.description != description
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_domain.py]: Domain was searched for by name, how could it not match here?

File: cloud/openstack/os_keystone_domain.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_keystone_domain
+short_description: Manage OpenStack Identity Domains
+extends_documentation_fragment: openstack
+version_added: "2.0"
+description:
+    - Manage  OpenStack Identity Domains
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_domain.py]: Docs should reflect that description and enabled can be modified on a domain that exists with the supplied name.

**Comment by gregdek** [comment]
@emonty ping for update (also seems to be failing travis)

**Comment by gregdek** [comment]
@emonty another ping

**Comment by emonty** [comment]
On 08/28/2015 10:24 AM, Greg DeKoenigsberg wrote:

> @emonty another ping
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ansible/ansible-modules-core/pull/1592#issuecomment-135838747

Back from vacation - will finish next week

**Comment by gregdek** [comment]
@emonty @shrews @dguerri @juliakreger @j2sol @rcarrillocruz ping again -- also this PR now fails Travis...

**Comment by Shrews** [comment]
@j2sol Addressed your comments with the latest commit.

@emonty This needs a few changes to merge into shade.

**Comment by omgjlk** [comment]
This looks good to me now. 

shipit (pending shade changes)

**Comment by gregdek** [comment]
So how do we know when the shade changes have been made? Waiting on @emonty for those?

**Comment by Shrews** [comment]
@gregdek The changes to shade have been made and merged. I think we're just waiting for @emonty to release the next shade version that will contain these changes, then he'll comment here, I suspect.

**Comment by emonty** [comment]
@gregdek new shade released - this should be good to go

**Comment by Shrews** [comment]
ping @bcoca @gregdek @jimi-c

**Comment by sivel** [comment]
This has been merged manually into ansible-modules-extras and `version_added` was bumped to 2.1

See https://github.com/ansible/ansible-modules-extras/compare/570869ef48ec...f798240f436a

**Pull request closed without merging**